pkg/outback/outback.go

```diff
@@ -38,13 +38,24 @@ type Outback struct {
 
 // New creates a Outback session and connects to AWS to create a session
 func New(awsConfig *AwsConfig) *Outback {
+	// default to -1 which defers the max retry to the service specific configuration.
+	// see: https://pkg.go.dev/github.com/aws/aws-sdk-go/aws#Config.MaxRetries
+	maxRetries := -1
+	if os.Getenv("AWS_MAX_ATTEMPTS") != "" {
+		m, err := strconv.Atoi(os.Getenv("AWS_MAX_ATTEMPTS"))
+		if err == nil {
+			maxRetries = m
+		}
+	}
+
```

Comment:
Thank you for adding this configuration for deployment attempts.

Is there a delay between retry attempts?

Also, is there any way we can get AWS to increase our limit similar to when we had to up the rate limit to work with CircleCI?

Reply:
This is using `aws.Client.DefaultRetryer` which will use a variable delay depending on the AWS error:

```go
const (
	// DefaultRetryerMinRetryDelay sets minimum retry delay
	DefaultRetryerMinRetryDelay = 30 * time.Millisecond

	// DefaultRetryerMinThrottleDelay sets minimum delay when throttled
	DefaultRetryerMinThrottleDelay = 500 * time.Millisecond

	// DefaultRetryerMaxRetryDelay sets maximum retry delay
	DefaultRetryerMaxRetryDelay = 300 * time.Second

	// DefaultRetryerMaxThrottleDelay sets maximum delay when throttled
	DefaultRetryerMaxThrottleDelay = 300 * time.Second
)
```
From: https://github.com/aws/aws-sdk-go/blob/4f0bc23ad379b199b41670feae94b026a90e64cb/aws/client/default_retryer.go#L38-L53

Some AWS throttle limits (like the ECS DescribeTaskDefinition) are global across the entire AWS account so AWS is hesitant to raise them again if they already raised it once (like they did for the CircleCI work)

Reply:
One last question/concern. 

If there are multiple concurrent deployments for the same environment, how often will we need to worry about a mismatch or issues with order in task definition versions?

Ex: 
Deployment 1 is at task version #100
Deployment 2 is at task version #101
Deployment 1 encounters a rate limit error and sleeps
Deployment 2 completes and the task definition is at #101
Deployment 1 completes and the task definition is at #100, and is missing changes from the merge that triggered Deployment 2

Reply:
@jclarkkoala great question! I agree, that is definitely a valid concern

I can think of two possible fixes: 
* configure the underlying CI to prevent parallel runs by canceling deploy 1 once deploy 2 starts
* update `outback` to have some external global lock that it can use internally to preserve the deploy order.

Based on those two I think it would be easier for this problem to be solved outside `outback` and at the CI level

That being said this edge case seems rare and infrequent (there's a lot of inherent randomness to deploys that might help avoid this scenario in most cases). So it will be interesting to see if we observe this in the real world.